wiki/misc/codecs/en.md

```diff
@@ -0,0 +1,171 @@
+- `Codec.DOUBLE`: A `double` codec.
+    - `Codec<Double> Codec.doubleRange(double min, double maxInc)`: An `double` codec with an inclusive range.
+- `Codec.STRING`: A `string` codec.
+
+"Ok", you tell me, "Thats cool. But... I still don't know what codecs are for or how to use them".
+
+### Basic Codec Example
+Let's go over a very basic example:
+
+```java
+boolean bool = 
+        Codec.BOOL.decode(
+            JsonOps.INSTANCE,
+            new JsonPrimitive(true)
+        )
+        .result()
+        .get()
+        .getFirst();
+
+assert bool;
+```
+
+WOAH! That doesn't look simple *at all*. What happened?
+
+Well, using `Codec`s is fairly verbose, but that means you get a lot of useful information to help with errors and such, which is important for Mojang to provide in their library since we want to know why Minecraft failed to load something, not just that it failed.
+
+Now, lets break this down into a couple sections.
+
+First off:
+```java
+Codec.BOOL.decode(
+    JsonOps.INSTANCE, // DynamicOps<T> ops
+    new JsonPrimitive(true) // T input
+) // DataResult<Pair<Boolean, JsonElement>>
+...
+```
+The `decode` method on a codec takes two values, an `ops` and an `input`.
+As shown in the comments above, the type of the input and a generic parameter on `ops` must match.
+This is because the `ops` needs to know about how the `input` functions.
+In this example, we use `com.mojang.serialization.JsonOps.INSTANCE`, which operates on json elements from `gson`.
+We then pass in a `JsonPrimative` with a value of `true` for this example.
+
+Finally, the `com.mojang.serialization.DataResult<Pair<A, T>>` type allows us to encode more information than just the result.
+First off, the `A` type is the output of the `Codec`, which is `Boolean` in this case, and the `T` is the same as the input.
+
+Let's look more into the `DataResult`:
+```java
+...
+.result() // Option<Pair<Boolean, JsonElement>>
+...
+```
+
+Ok, this starts to make more sense. 
+`DataResult` has a lot of associated methods on it, but for now let's only cover two: `result` and `error`.
+`error` returns a `PartialResult`, which allows you to both recover a decode, and to get the error message for why the decode failed. Right now, the `result` method is more important to us.
+`result` returns an `Option<Pair<A, T>>`, which makes sure that we know for sure if we have a result, otherwise we could just get null.
+
+Finally, we get to the last two lines:
+```java
+...
+.get() // Pair<Boolean, JsonElement>
+.getFirst(); // Boolean
+```
+
+We use `get` to unbox the `Option`. Generally this is unsafe to do, an IntelliJ even gives a warning.
+In this case we know that it is safe due to the simplicity of the example.
+Then finally, we call `getFirst` on `com.mojang.datafixers.util.Pair` to get the first half of the pair
+
+Wow. That sure was a lot.
+Now, I know this may seem like the `Codec` system is complicated right now, but unfortunately we have only scratched the surface.
```

Comment:
Using the first person in a wiki article feels a bit weird to me, especially given that they end up written by several people. In this case, I think "Now, this may seem" would be enough.